### PR TITLE
New scorers

### PR DESCRIPTION
**Pull request by cservan**
Add new scorer to evaluate valid file while training.
these scorers are : Translation Edit Rate (TER), Word Error Rate (WER), f-measure, precision and recall (usefull for the seq2seq tagging model)

### CODE REVIEW

File: opennmt/utils/wer.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
+"""Hypotheses file scoring."""
+import numpy
+import pyter
+
+def wer(ref_path, hyp_path):
+  """ Compute Word Error Rate between two files """
+  ref_fp = open(ref_path)
+  hyp_fp = open(hyp_path)
```
Comment (guillaumekln) [inline opennmt/utils/wer.py:8]: These file handles should be closed at the end of the function. To do it automatically, you could turn this into:

```python
with open(ref_path) as ref_fp, open(hyp_path) as hyp_fp:
  ...
```

This also applies for the other scoring functions.

File: opennmt/utils/scorers.py [hunk 2]
```diff
@@ -95,6 +170,18 @@ def make_scorers(names):
...
+    elif name == "wer":
+      scorer = WERScorer()
+    elif name == "ter":
+      scorer = TERScorer()
+    elif name == "precision":
+      scorer = PRECISIONScorer()
+    elif name == "recall":
+      scorer = RECALLScorer()
+    elif name == "fmeasure":
+      scorer = FMEASUREScorer()
```
Comment (guillaumekln) [inline opennmt/utils/scorers.py:182]: I'm wondering if we need these scores as `PRFScorer` computes all values in one shot. What do you think?

File: setup.py [hunk 3]
```diff
@@ -48,7 +48,8 @@ def get_long_description():
         "rouge>=1.0,<2",
         "sacrebleu>=1.4.3,<2",
         "tensorflow>=2.1,<2.2",
-        "tensorflow-addons>=0.8.1,<0.9"
+        "tensorflow-addons>=0.8.1,<0.9",
+        "pyter3>=0.3"
```
Comment (guillaumekln) [inline setup.py:52]: As we don't know how stable this package is, let's use a strict requirement `pyter3==0.3`.

File: opennmt/utils/wer.py (hunk 1 above)
Comment (cservan) [reply opennmt/utils/wer.py:8]: Good suggestion, fine by me

File: opennmt/utils/scorers.py (hunk 2 above)
Comment (cservan) [reply opennmt/utils/scorers.py:182]: Actually, it was to have separate scores only, but yes, the last scorer could be used.

File: setup.py (hunk 3 above)
Comment (cservan) [reply setup.py]: Done.

**Pull request merged**